pandas/tests/io/formats/style/test_html.py

```diff
@@ -1003,3 +1013,84 @@ def test_to_html_na_rep_non_scalar_data(datapath):
 </table>
 """
     assert result == expected
+
+
+@pytest.mark.parametrize("escape_axis_0", [True, False])
+@pytest.mark.parametrize("escape_axis_1", [True, False])
+def test_format_index_names(styler_multi, escape_axis_0, escape_axis_1):
+    if escape_axis_0:
+        styler_multi.format_index_names(axis=0, escape="html")
+        expected_index = ["X&gt;", "y_"]
+    else:
+        expected_index = ["X>", "y_"]
+
+    if escape_axis_1:
+        styler_multi.format_index_names(axis=1, escape="html")
+        expected_columns = ["A&amp;", "b&amp;"]
+    else:
+        expected_columns = ["A&", "b&"]
+
+    result = styler_multi.to_html(table_uuid="test")
+    expected = f"""\
+<style type="text/css">
+</style>
+<table id="T_test">
+  <thead>
+    <tr>
+      <th class="blank" >&nbsp;</th>
+      <th class="index_name level0" >{expected_columns[0]}</th>
+      <th id="T_test_level0_col0" class="col_heading level0 col0" colspan="2">A</th>
+      <th id="T_test_level0_col2" class="col_heading level0 col2" colspan="2">B</th>
+    </tr>
+    <tr>
+      <th class="blank" >&nbsp;</th>
+      <th class="index_name level1" >{expected_columns[1]}</th>
+      <th id="T_test_level1_col0" class="col_heading level1 col0" >a</th>
+      <th id="T_test_level1_col1" class="col_heading level1 col1" >b</th>
+      <th id="T_test_level1_col2" class="col_heading level1 col2" >a</th>
+      <th id="T_test_level1_col3" class="col_heading level1 col3" >b</th>
+    </tr>
+    <tr>
+      <th class="index_name level0" >{expected_index[0]}</th>
+      <th class="index_name level1" >{expected_index[1]}</th>
```

Comment:
Its probably sufficient to assert `expected` in `result` rather than equate the large bulk of text. This also future proofs the test so that if other changes are implemented which dont affect the specific format_index_names function this test wont need updating.

Reply:
updated